components/ai_chat/core/browser/ai_chat_metrics.cc

```diff
@@ -349,6 +513,83 @@ void AIChatMetrics::ReportChatCounts() {
   // TODO(djandries): remove the following report when Nebula experiment is over
   p3a_utils::RecordToHistogramBucket(kChatCountNebulaHistogramName,
                                      kChatCountBuckets, chat_count);
+
+  uint64_t max_chat_duration =
+      chat_durations_storage_.GetHighestValueInPeriod();
+  p3a_utils::RecordToHistogramBucket(kMaxChatDurationHistogramName,
+                                     kMaxChatDurationBuckets,
+                                     max_chat_duration);
+
+  uint64_t chat_with_history_count =
+      chat_with_history_count_storage_.GetWeeklySum();
+  int history_percentage = static_cast<int>(std::ceil(
+      chat_count > 0
+          ? (static_cast<double>(chat_with_history_count) / chat_count) * 100.0
+          : 0.0));
+
+  p3a_utils::RecordToHistogramBucket(kChatHistoryUsageHistogramName,
+                                     kChatHistoryUsageBuckets,
+                                     history_percentage);
+}
+
+void AIChatMetrics::RecordContextSource(ConversationHandlerForMetrics* handler,
+                                        mojom::ConversationTurnPtr& entry) {
+  ContextSource context = ContextSource::kTextInputWithoutPage;
+  if (prompted_via_omnibox_) {
+    context = ContextSource::kOmniboxInput;
+  } else if (entry->text ==
+                 l10n_util::GetStringUTF8(IDS_CHAT_UI_SUMMARIZE_PAGE) ||
+             entry->text ==
+                 l10n_util::GetStringUTF8(IDS_CHAT_UI_SUMMARIZE_VIDEO)) {
+    context = ContextSource::kPageSummary;
+  } else if (handler->GetConversationHistorySize() == 1 &&
+             handler->IsConversationStarter(entry->text)) {
+    UMA_HISTOGRAM_BOOLEAN(kUsedConversationStarterHistogramName, true);
+    context = ContextSource::kConversationStarter;
+  } else if (entry->action_type != mojom::ActionType::UNSPECIFIED &&
+             entry->action_type != mojom::ActionType::RESPONSE &&
+             entry->action_type != mojom::ActionType::QUERY) {
+    context = ContextSource::kRightClick;
```

Comment:
this seems unreliable - if we add action types but forget to check here, then we could artificially inflate right-click sources

Reply:
any suggestions on how to improve this? if we change this to an allowlist, then we could miss out on newly added quick actions

Reply:
Perhaps in `RecordContextMenuUsage` mark similarly to `prompted_via_omnibox_` so that we know definitively when it's from context menu, and then have a `ContextSource::Unknown` for anything not known at this time.